strategizer/strategizer.js

```diff
@@ -51,11 +52,12 @@ export class Strategizer {
     let generation = this.generation + 1;
     let individualsPerStrategy = Math.floor(this._options.generationSize / this._strategies.length);
     let generated = await Promise.all(this._strategies.map(strategy => {
+      let recipeFilter = recipe => this._ruleset.isAllowed(strategy, recipe);
       return strategy.generate({
         generation: this.generation,
-        generated: this.generated,
-        terminal: this.terminal,
-        population: this.population,
+        generated: this.generated.filter(recipeFilter),
+        terminal: this.terminal.filter(recipeFilter),
+        population: this.population.filter(recipeFilter),
```

Comment:
How come you're filtering generated and terminal here too?

Reply:
This is what we give to the strategy as input and what strategy operates on.

Most strategies operate on 'generated', though some also work on 'terminal' (e.g. SearchTokensToParticles, FallbackFate).

The idea here is to restrict what the strategy is given to only those recipes that the strategy is allowed to create new descendants of.